### PR TITLE
[feature] New command to run all possible migrations 

### PR DESCRIPTION
**Pull request by shedninja6**
### Summary
Created an interface for current and future fogg migrations. The interface allows all migrations to run in a single place. The interface requires that all migrations support Guard and Migrate methods. Consolidated all migrations (versioning and file type into a single folder)

- `fogg migrate` defaults to fogg.yml; however, a fogg.json can be provided. 
- Also improved logging information, to give more specific user info

### Test Plan
- Added yaml_test.go to test migrating from fogg.json to fogg.yml (~85% code coverage). 
- Updated upgrade_test.go to support new interface (~85% code coverage).

### CODE REVIEW

File: cmd/migrate.go [hunk 1]
```diff
@@ -4,20 +4,21 @@ import (
...
+	"github.com/chanzuckerberg/fogg/migrations"
 	"github.com/spf13/afero"
 	"github.com/spf13/cobra"
 )
 
 func init() {
-	yamlMigrateCmd.Flags().StringP("config", "c", "fogg.json", "Use this to override the fogg config file.")
-	rootCmd.AddCommand(yamlMigrateCmd)
+	migrateCommand.Flags().StringP("config", "c", "fogg.json", "Use this to override the fogg config file.")
+	migrateCommand.Flags().BoolP("skip", "s", false, "Use this to run all tests.")
```
Comment (edulop91) [inline cmd/migrate.go]: a bit unclear on what `skip` does from the description only

File: cmd/migrate.go [hunk 2]
```diff
@@ -4,20 +4,21 @@ import (
...
+	rootCmd.AddCommand(migrateCommand)
 }
 
-var yamlMigrateCmd = &cobra.Command{
-	Use:   "yaml-migrate",
+var migrateCommand = &cobra.Command{
+	Use:   "migrate",
 	Short: "Converts existing fogg.json to fogg.yml",
-	Long:  "This command will convert the fogg.json to a fogg.yml file type.",
+	Long:  "This command will run all applicable updates to fogg.",
```
Comment (edulop91) [inline cmd/migrate.go]: `to fogg configuration`?

File: config/config.go [hunk 3]
```diff
@@ -43,14 +43,15 @@ func InitConfig(project, region, bucket, table, awsProfile, owner, awsProviderVe
 		Docker:   false,
 		Envs:     map[string]v2.Env{},
 		Modules:  map[string]v1.Module{},
+		Version:  2,
```
Comment (edulop91) [inline config/config.go]: maybe this should be a separate pr

File: migrations/migrations.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+		shouldRun, err := migration.Guard(fs, configFile)
+		if err != nil {
+			return err
+		}
+		if shouldRun == false {
+			continue
+		}
+
+		//Ignores prompts if user chose to run all tests
+		if skipAll == false {
```
Comment (edulop91) [inline migrations/migrations.go]:
```
if force || migration.prompt() {
  continue
}
```

File: cmd/migrate.go (hunk 1 above)
Comment (edulop91) [reply cmd/migrate.go]: maybe better as `force`? idk what you think

File: migrations/upgrade_test.go [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+	r.Equal(true, shouldRun)
+
+	configFile, err := upgradeMigration.Migrate(fs, confPath)
+	r.NoError(err)
+
+	//configFile is passed to validate the Migrate returned string
+	_, v, err = config.FindConfig(fs, configFile)
+	r.NoError(err)
+	r.Equal(2, v) // now upgraded
+}
```
Comment (edulop91) [inline migrations/upgrade_test.go]:
```suggestion

}
```

File: migrations/yaml_test.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	fs, _, err := util.TestFs()
+	r.Nil(err)
+
+	err = afero.WriteFile(fs, confPath, json, 0644)
+	r.Nil(err)
+
+	shouldRun, err := yamlMigration.Guard(fs, confPath)
+	r.Nil(err)
+	r.Equal(false, shouldRun)
+}
```
Comment (edulop91) [inline migrations/yaml_test.go]:
```suggestion

}
```

File: migrations/migrations.go [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+		&VersionUpgradeMigration{}, //This runs first because v1 to v2 uses json.Marshal
+		&JSONToYamlMigration{},
+	}
+
+	for _, migration := range migrations {
+		shouldRun, err := migration.Guard(fs, configFile)
+		if err != nil {
+			return err
+		}
+		if shouldRun == false {
```
Comment (edulop91) [inline migrations/migrations.go]: maybe this could use a `log.Debug()` or something to give feedback that you've skipped a migration

File: cmd/migrate.go [hunk 8]
```diff
@@ -32,8 +33,13 @@ var yamlMigrateCmd = &cobra.Command{
 			return errs.WrapInternal(err, "couldn't parse config flag")
 		}
 
+		skipPrompts, err := cmd.Flags().GetBool("skip")
```
Comment (edulop91) [inline cmd/migrate.go]: this is now named `force`

File: migrations/migrations.go [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
```
Comment (mbarrien) [inline migrations/migrations.go]: Not a hard and fast rule, but I'd probably remove the redundant comparison to a bool and just write "if !shouldRun". Same with "if skipPrompts || !migration.Prompt()" a few lines below.
